salt/modules/cmdmod.py

```diff
@@ -2974,43 +2977,62 @@ def _cleanup_tempfile(path):
                 "stderr": "",
                 "cache_error": True,
             }
-        shutil.copyfile(fn_, path)
+        try:
+            shutil.copyfile(fn_, path)
+        except FileNotFoundError:
+            _cleanup_tempfile(path)
+            # If a temp working directory was created (Windows), let's remove that
+            if win_cwd:
+                _cleanup_tempfile(cwd)
+            return {
+                "pid": 0,
+                "retcode": 1,
+                "stdout": "",
+                "stderr": "",
+                "cache_error": True,
+            }
     if not salt.utils.platform.is_windows():
         os.chmod(path, 320)
         os.chown(path, __salt__["file.user_to_uid"](runas), -1)
 
-    if salt.utils.platform.is_windows():
-        if shell.lower() not in ["powershell", "pwsh"]:
-            cmd_path = _cmd_quote(path, escape=False)
-        else:
-            cmd_path = path
+    if isinstance(args, (list, tuple)):
+        new_cmd = [path, *args] if args else [path]
     else:
-        cmd_path = _cmd_quote(path)
+        new_cmd = [path, str(args)] if args else [path]
```

Comment:
This does not look correct. This appears to be taking a str of potentially multiple arguments and passing it as just the first argument.

Also the docstring only states that a str is even supported and not a list/tuple

```
:param str args: String of command line args to pass to the script. Only
        used if no args are specified as part of the `name` argument. To pass a
string containing spaces in YAML, you will need to doubly-quote it.
```

Reply:
Thanks. This should be fixed with https://github.com/saltstack/salt/pull/68301/files#diff-700b7d75700c9eafa3ee48d01236e59f37b2ed81d998267136fbcc0224766497R3007 , althrough passing multiple arguments with a single string might be unreliable, especially on Windows.